### ISSUE TITLE
Add setting to choose which attributemap to use, oid2name

### ISSUE THREAD

**Issue by danowar2k** [catalyst/moodle-auth_saml2#384]
We have a custom LDAP attribute ("urn:oid:...") which is not covered by the SimpleSAMLphp attributemap, probably the one in here:
https://github.com/catalyst/moodle-auth_saml2/blob/master/extlib/simplesamlphp/attributemap/oid2name.php

There doesn't seem to be a GUI way to add custom oid to name mappings. 
Is there a recommended way to go about this? I don't know if that file would be the only place that needed to be edited.

Otherwise, it would be really nice to be able to extend the attributemaps somehow as the current way really limits what we can do with SAML.

**Comment by brendanheywood** [comment]
In the saml attribute to moodle user profile field mappings you can just paste the long urn:oid... string and that should just work

**Comment by danowar2k** [comment]
Yeah we discovered that too, but then the readability of that setting is not very high, as all the urn:oid stuff is not the important stuff, the suffix is.

**Comment by brendanheywood** [comment]
hi @danowar2k the attribute map is set here:

https://github.com/catalyst/moodle-auth_saml2/blob/master/config/config.php#L94-L97

and it is selected from the list of build in maps here:

https://github.com/catalyst/moodle-auth_saml2/tree/master/extlib/simplesamlphp/attributemap

Is the mapping you want one of the ones included?

Converting that from a hardcoded oid2name to a select dropdown of all the files in that directory would be fairly easy. If you wanted to open a pull request for that it would be very welcome.

The alternative workaround would be to make a new file in that directory and then swap the config over to use it.

**Comment by danowar2k** [comment]
The problem is that we have defined custom (!) attributes, meaning they are not in any of your maps (of course). So the request wasn't to be allowed to change the used attribute maps but to allow one to store new custom mappings in the database. (e.g. add "oid:XXX:NEVER_BEFORE_SEEN_FIELD" -> "NEVER_BEFORE_SEEN_FIELD")

**Comment by brendanheywood** [comment]
So if you have defined your own attributes then that's fine. With many IdP 's (eg adfs) you can re-map attributes to something specific for each SP so that is one way. Or you can just use the mappings gui which is pretty well exactly what this is for. So I think think boils down to pure a UX issue then because the field is too short to show the value? Or does it not actually work because it fails form validation?

As these fields are all managed by core and common to all auth plugins so there isn't much I can do about it in this plugin itself, probably needs to be a tracker

**Comment by brendanheywood** [comment]
Depending on what you're attribute is called this setting may help:

![image](https://user-images.githubusercontent.com/187449/75038756-2ca16f80-550b-11ea-9708-093efcdd6a17.png)

**Comment by danowar2k** [comment]
Thanks for answering, my lag is incredible at the moment ;-)

I think we can close this here...
